server/src/main/java/com/broadcom/lsp/cobol/service/CobolTextDocumentService.java

```diff
@@ -256,6 +264,17 @@ public void observerCallback(@NonNull RunAnalysisEvent event) {
     docs.forEach((key, value) -> analyzeDocumentFirstTime(key, value.getText(), event.verbose));
   }
 
+  @Override
+  @CheckServerShutdownState
+  public CompletableFuture<List<Location>> ranges(@NonNull JsonObject json) {
```

Comment:
In the latest discussion with @ishche, we came up with an idea to use a slightly different API: CCF provides URI and expects an instance of `programUnit`. Here is type description in TypeScript format:
```
type paragraph = {
    name: string
    section: string
    location: {
        uri: string
        start: {
            line: number
            character: number
        }
        end: {
            line: number
            character: number
        }
    }
    usages: [{
        uri: string
        start: {
            line: number
            character: number
        }
        end: {
            line: number
            character: number
        }
    }]
}
type programUnit = {
    programID: string
    procedureDivision: {
        paragraphs: paragraph[]
        location: {
            uri: string
            start: {
                line: number
                character: number
            }
            end: {
                line: number
                character: number
            }
        }
    }
}
```
Or course we can construct it on the client, not here.

Reply:
As discussed I have changed the return object to AnalysisResult